stock_picking_actual_date/models/stock_scrap.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2024 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class StockScrap(models.Model):
+    _inherit = "stock.scrap"
+
+    actual_date = fields.Date(tracking=True)
+    is_editable_actual_date = fields.Boolean(
+        compute="_compute_is_editable_actual_date", string="Is Editable"
+    )
+
+    def _compute_is_editable_actual_date(self):
+        for rec in self:
+            rec.is_editable_actual_date = False
+            if rec.state != "done" or self.env.user.has_group(
+                "stock.group_stock_manager"
+            ):
+                rec.is_editable_actual_date = True
```

Comment:
Do you think it's a good idea to create a mixin and let stock.picking and stock.scrap inherit it?